_posts/2023-02-15-my-first-patch-set.md

```diff
@@ -0,0 +1,153 @@
+- `-Wmissing-prototypes`
+- Excess arguments on Kernel docs
+
+At first the set was composed only by 4 patches, one for each type of warning, but after some advice
+from [Maíra Canal][maira-canal] they were split even more as different tactics were used to fix the
+same type of problem in different places.
+
+## How I Sent
+
+For a more in depth procedure, please read the [kernel documentation about this
+topic][kernel-doc-applying-patch].
+
+First, i used the `check-patch` utility to check all the commits for any code style error
+
+```bash
+./scripts/check-patch --strict -g HEAD..drm-misc/drm-misc-next
+```
+
+Arguments:
+- `--strict`: enable more subjective tests
+- `-g HEAD..drm-misc/drm-misc-next`: check all the commit until the drm-misc-next branch
+
+To make the patch set i used the git format-patch utility
+```bash
+# On the fix-warning branch
+git format-patch drm-misc/drm-misc-next -o outgoing --cover-letter
+```
+
+Arguments:
+- `drm-misc/drm-misc-next`: get all the commits until the drm-misc-next branch
+- `-o outgoing`: place the patches on the outgoing directory
+- `--cover-letter`: generate a first patch file with a cover letter to write about the whole patch
+
+After I wrote my cover letter and made various sanity checks such as: compiling the kernel one more
+time and reading the patch files to spot any errors. I used the `get_maintainer` utility to find the
+emails and mail list that i had to send my patches.
+
+```bash
+./scripts/get_maintainer.pl outgoing/*
+```
+
+Arguments:
+- `outgoing/*`: read all the patches inside the `outgoing` directory
+
+After that I used the `git send-email` to send the patches to all the emails that were maintainers
+or supporters on the `get_maintainer` output and to the dri-devel and amd-gfx mailing lists.
+
+```bash
+git send-email \
+--to='amd-gfx@lists.freedesktop.org, dri-devel@lists.freedesktop.org' \
+--cc='<emails-of-maintainers-and-subscribers>' outgoing/*
+```
+
+To a tutorial on how to setup git send email, I recommend the [sourcehut
+tutorial][sourcehut-tutorial].
+
+After all this work I needed to wait for responses and hope that any dumb errors were present :).
+
+## How the Patch was Received
+
+Fortunately the patches had a pretty good response!
+
+The [first version of the patch set][first-patch] was with sent 10 patches. Six of them were
+accepted with no requests for changes. The other four received different responses. One was thought
+unnecessary, other hasn't had any response yet, and the other two were requested some modification.
+
+Those last couple ones will be sent on the second version of this patch.
+
+##  What To Do Next
+
+After this first version I'll send a v2 with the 2 patches that were asked revision and the pathc
+that did not receive any response on the v1.
+
+After that I pretend to add more unit test  with [Kunit][kunit] on the DRM side. I will start by
+running the kernel code coverage tool Gcov. But this will be explained better on a next blog post.
```

Comment:
```suggestion
After that, I pretend to keep preparing myself for GSoC 2023! I'm thinking of exploring the unit test on the DRM side with  [Kunit][kunit] on the DRM side. But this will be explained better in the next blog post.
```